### PR TITLE
[WIP]Feature/arm9 validation for registration

### PR DESCRIPTION
**Pull request by HerrX2000**
Added server-side and client-side _**feedback for the user**_ for validation. (Checks allready merged)

### CODE REVIEW

**Comment by HerrX2000** [comment]
I guess inline script isn't allowed either. I am working on a userfeedback option, that doesn't rely on js alerts.

File: templates/user/register.html.twig [hunk 1]
```diff
@@ -33,5 +60,10 @@
             </form>
 
          </div>
+         <script>
+        $(document).ready(function(){
```
Comment (flaver12) [inline templates/user/register.html.twig]: Don't use query there a native js functions for that

File: templates/user/register.html.twig [hunk 2]
```diff
@@ -6,12 +6,38 @@
 {% block body_title %}<h1>ARM/{{ controller_name }} </h1>{% endblock %}
 
 {% block body_content %}
+
+<script>
+function validateForm() {
```
Comment (flaver12) [inline templates/user/register.html.twig]: Add this to our js lib, so we could use that at other points

File: templates/user/register.html.twig (hunk 1 above)
Comment (HerrX2000) [reply templates/user/register.html.twig]: Has been removed. Wasn't needed anyway.

File: templates/user/register.html.twig (hunk 2 above)
Comment (HerrX2000) [reply templates/user/register.html.twig]: Added it to app.js but doesn't work.
register:58 Uncaught ReferenceError: validateForm is not defined
    at HTMLFormElement.onsubmit (register:58)

File: assets/js/app.js [hunk 3]
```diff
@@ -76,4 +76,36 @@ app.addComponent('http', new httpComponent(app));
 app.run();
 
 //Access for inline scripts
-window.app = app;
\ No newline at end of file
+window.app = app;
+
+//Form validation
+
+function validateForm(formName) {
```
Comment (flaver12) [inline assets/js/app.js]: One more improvement can be done. Add it as a module. You can simple look at the ajax/http component.

Ask me if you need there help

File: src/Controller/UserController.php [hunk 4]
```diff
@@ -28,21 +28,50 @@ public function register(Request $request)
...
         else{
-            $password=$request->query->get('password');
-            $password_val=$request->query->get('password_validation');
-            $user=$request->query->get('username');
-            $email=$request->query->get('email');
+            return new Response('Username cant be empty');
         }
         $entityManager->flush();
-        if(strlen($user) <= 20 and strlen($password) >= 8 and filter_var($email, FILTER_VALIDATE_EMAIL) and $password==$password_val){
+        if(strlen($user) <= 20){
```
Comment (flaver12) [inline src/Controller/UserController.php]: Mabye do it as a Validation class. I will make a example in your branch, so that you can implement the rest :)

Comment (HerrX2000) [reply src/Controller/UserController.php]: Sure good point. Working on that.

File: assets/js/app.js (hunk 3 above)
Comment (HerrX2000) [reply assets/js/app.js]: So I sperate validateForm from app.js and create a new component?

Comment (HerrX2000) [reply assets/js/app.js]: Did quite get what you meant. The new commit has the valForm function in a seperat js file and works now. But doesn't get 'build' by yarn.

**Pull request closed without merging**